src/Uno.UWP/Storage/Streams/DataReader.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Windows.Storage.Streams
+{
+	public partial class DataReader : IDataReader, IDisposable
+	{
+		private IBuffer _buffer = null;
+		private uint _position = 0;
+
+		public ByteOrder ByteOrder { get; set; }
+
+		public uint UnconsumedBufferLength
+		{
+			get => _buffer.Length - _position;
+		}
+		//public DataReader(global::Windows.Storage.Streams.IInputStream inputStream)
+		//{
+		//	global::Windows.Foundation.Metadata.ApiInformation.TryRaiseNotImplemented("Windows.Storage.Streams.DataReader", "DataReader.DataReader(IInputStream inputStream)");
+		//}
+
+		private void CheckPosition(int size)
+		{
+			if (_buffer.Length >= (_position + size))
+			{
+				return;
+			}
+
+			throw new IndexOutOfRangeException("Windows.Storage.Streams.DataReader - reading past EOF");
+		}
+
+		public byte ReadByte()
+		{
+			CheckPosition(1);
+			byte value = _buffer.Data[_position];
+			_position++;
+			return value;
+		}
+
+		public IBuffer ReadBuffer(uint length)
+		{
+			CheckPosition((int)length);
+
+			var buffer = new InMemoryBuffer((int)length);
+			for (int i = 0; i < length; i++)
+			{
+				buffer.Data[i] = _buffer.Data[_position];
+				_position++;
+			}
+			return buffer;
+		}
+		public bool ReadBoolean()
+		{
+			byte value = ReadByte();
+			if (value == 0)
+			{
+				return false;
+			}
+			return true;        // although DataWriter uses 0/1, we can also interpret all non-zero values as true;

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5527039)

Reply:
this is not code